### PR TITLE
Adds local storage functionality

### PR DESCRIPTION
**Pull request by cwefso**
This allows the user to 
- Add something to storage.
- Delete something and have it stay deleted on page reload.
- Allow it star/unstar an idea and have it stay that way on reload.

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -11,7 +11,8 @@ var background = document.querySelector('.background');
 
 var savedIdeas = [];
 
-// window.onload = retrieveMadeIdeaCards;
+window.onload = retrieveMadeIdeaCards;
```
Comment (cwefso) [inline src/main.js:14]: This pulls the array savedIdeas from localStorage and parses it out of being a string. It happens every time we refresh the page.

File: src/main.js [hunk 2]
```diff
@@ -46,6 +47,7 @@ function toggleStarOnData() {
       savedIdeas[i].star = true;
     }
   }
+  saveIdeaToStorage()
```
Comment (cwefso) [inline src/main.js:50]: Updates the savedIdeas array and saves this new array to localStorage, overwriting what is currently saved there.

File: src/main.js [hunk 3]
```diff
@@ -63,6 +65,7 @@ function toggleStarOffData() {
       savedIdeas[i].star = false;
     }
   }
+  saveIdeaToStorage()
```
Comment (cwefso) [inline src/main.js:68]: Updates the savedIdeas array and saves this new array to localStorage, overwriting what is currently saved there.

File: src/main.js [hunk 4]
```diff
@@ -79,6 +82,7 @@ function deleteFromArray(event) {
       savedIdeas.splice(savedIdeas.indexOf(savedIdeas[i]));
     }
   }
+  saveIdeaToStorage()
 }
```
Comment (cwefso) [inline src/main.js:86]: Updates the savedIdeas array and saves this new array to localStorage, overwriting what is currently saved there.

File: src/main.js [hunk 5]
```diff
@@ -122,20 +126,19 @@ function createNewIdea() {
   var currentIdea = new Idea(userNewTitle.value, userNewBody.value);
   if (userNewTitle.value && userNewBody.value) {
   savedIdeas.push(currentIdea);
+  saveIdeaToStorage();
```
Comment (cwefso) [inline src/main.js:129]: Updates the savedIdeas array and saves this new array to localStorage, overwriting what is currently saved there.

**Review by carlymclift — APPROVED** [review]
Thanks for adding those comments in there!

File: src/main.js [hunk 6]
```diff
@@ -122,20 +126,19 @@ function createNewIdea() {
...
-// function saveIdeaToStorage() {
-//   localStorage.setItem('ideas', JSON.stringify(savedIdeas));
-// }
-//
-// function retrieveMadeIdeaCards() {
-//   savedIdeas = JSON.parse(localStorage.getItem('ideas')) || [];
-//
-//   showUsersIdeaCard();
-// }
+function saveIdeaToStorage() {
```
Comment (cwefso) [inline src/main.js:134]: These are the functions saving to storage and retrieving from storage. The retrieve function only fire on page reload. The storing function happens each time we make a change to our data model (the savedIdeas array).

**Pull request merged**